File: Domain/ActionPointCard.cs
namespace Domain
{
    public class ActionPointCard
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public int UpgradeLevel { get; set; } = 0;
        public ICollection<ActionPointLevel> ActionPointLevels { get; set; } = new List<ActionPointLevel>();
    }
}

File: client-app/src/app/models/actionPointCard.ts
export interface ActionPointCard {
    id: string
    name: string
    upgradeLevel: number
    equippedBy: string | null
    actionPointLevels: Array<ActionPointLevel>
}

export interface ActionPointLevel {
    id: string
    level: number
    range: string
    cost: string
    duration: string
    prerequisite: string
    upgradeCost: string
    castingTime: string
    description: string
}